File: 53-MaximumSubarray/53-MaximumSubarray.java
// Last updated: 7/14/2026, 2:50:51 PM
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = 0;
        int max = Integer.MIN_VALUE;
        for(int i = 0; i <= nums.length - 1; i++){
            sum += nums[i];
            if(sum > max){
                max = sum;
            }
            if(sum < 0){
                sum = 0;
            }
        }
        return max;
    }
}

File: 53-MaximumSubarray/53-MaximumSubarray.js
// Last updated: 7/14/2026, 2:50:56 PM
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let sum = 0;
    let maxi = nums[0];

    for(let i = 0; i < nums.length; i++){
        sum += nums[i];

    maxi = Math.max(maxi , sum)

    if(sum < 0){
        sum=0;
    }
    }
    return maxi;

};